core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -486,12 +487,14 @@ private[kafka] class Processor(val id: Int,
   private def processCompletedReceives() {
     selector.completedReceives.asScala.foreach { receive =>
       try {
-        val channel = selector.channel(receive.source)
+        val openChannel = selector.channel(receive.source)
+        val channel = if (openChannel != null) openChannel else selector.closedChannel(receive.source)
```

Comment:
Nit: this may be a little better like:

```scala
val socketAddress = {
  val channel = if (openChannel != null) openChannel else selector.closedChannel(receive.source)
  channel.socketAddress
}
```

The reason is that some methods are unsafe to call on a closed channel. This way we make it clear that we are only calling the safe `socketAddress` on a potentially closed channel. Does that make sense?